carma_wm/src/collision_detection.cpp

```diff
@@ -268,25 +270,25 @@ namespace carma_wm {
                 }
             }
 
-            std::vector<point_t> unioin_future_polygon_points;
-            unioin_future_polygon_points.reserve(union_polygon_size);
+            std::vector<point_t> union_future_polygon_points;
+            union_future_polygon_points.reserve(union_polygon_size);
 
             for (auto i : op.fp){
                 if( std::get<0>(i) <= target_time) {
-                    unioin_future_polygon_points.insert( unioin_future_polygon_points.end(), std::get<1>(i).outer().begin(), std::get<1>(i).outer().end());
+                    union_future_polygon_points.insert( union_future_polygon_points.end(), std::get<1>(i).outer().begin(), std::get<1>(i).outer().end());
                 }
             }
 
             polygon_t union_polygon;  
-            boost::geometry::assign_points(union_polygon, unioin_future_polygon_points);
+            boost::geometry::assign_points(union_polygon, union_future_polygon_points);
 
             polygon_t hull_polygon;
             boost::geometry::convex_hull(union_polygon, hull_polygon);
 
-            collision_detection::MovingObject output_object = {hull_polygon, op.linear_velocity};
+            collision_detection::MovingObject output_object = {hull_polygon, op.linear_velocity}; //TODO: fix missing initializer for fp element
```

Comment:
Not quire sure what this TODO comment is referring too

Reply:
The MovingObject contains 3 elements, and only two are initialized here.  fp also needs to be initialized.  This is a complex object itself, without any documentation, so I'm not sure what its purpose is.

Reply:
I tracked down the meaning of fp (and added description!) and made a few changes to fix this warning in several places.